xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/viewers/attachments.js

```diff
@@ -218,29 +218,46 @@ require(['jquery', 'xwiki-events-bridge'], function($) {
       }
     })
   });
+  $(document).on("xwiki:livetable:loadingComplete", function(event, data) {
+    if (data.tableId == 'docAttachments') {
+      $(document).on('click', '#docAttachments .actiondelete', function(event) {
+        event.preventDefault();
+        $('#deleteAttachment').data('relatedTarget', event.currentTarget);
+        $('#deleteAttachment').modal('show');
+      });
+
+      updateCount();
```

Comment:
There might be a way to avoid doing another request to get the attachment count, at least in this case.